engine/src/main/java/org/hibernate/search/store/impl/DefaultLockFactoryCreator.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.store.impl;
+
+import java.io.File;
+import java.util.Properties;
+
+import org.apache.lucene.store.LockFactory;
+import org.apache.lucene.store.NativeFSLockFactory;
+import org.apache.lucene.store.NoLockFactory;
+import org.apache.lucene.store.SimpleFSLockFactory;
+import org.apache.lucene.store.SingleInstanceLockFactory;
+import org.hibernate.search.cfg.Environment;
+import org.hibernate.search.engine.service.spi.ServiceManager;
+import org.hibernate.search.exception.SearchException;
+import org.hibernate.search.spi.BuildContext;
+import org.hibernate.search.store.LockFactoryProvider;
+import org.hibernate.search.store.spi.LockFactoryCreator;
+import org.hibernate.search.util.impl.ClassLoaderHelper;
+
+/**
+ * The one and only {@link LockFactoryCreator}.
+ *
+ * @author Sanne Grinovero
+ * @author Hardy Ferentschik
+ * @author Gunnar Morling
+ */
+public class DefaultLockFactoryCreator implements LockFactoryCreator {
+
+	private ServiceManager serviceManager;
+
+	@Override
+	public void start(Properties properties, BuildContext context) {
+		this.serviceManager = context.getServiceManager();
+	}
+
+	@Override
+	public LockFactory createLockFactory(File indexDir, Properties dirConfiguration) {
+		//For FS-based indexes default to "native", default to "single" otherwise.
+		String defaultStrategy = indexDir == null ? "single" : "native";
+		String lockFactoryName = dirConfiguration.getProperty( Environment.LOCKING_STRATEGY, defaultStrategy );
+		if ( "simple".equals( lockFactoryName ) ) {
+			if ( indexDir == null ) {
+				throw new SearchException( "To use \"simple\" as a LockFactory strategy an indexBase path must be set" );
```

Comment:
I just took this code as is from the original class. I'll add a commit for using JBL.